Fix inconsistent hover color on Masterbar

okmttdhr opened the pull request:
Fix https://github.com/Automattic/dotcom-forge/issues/6798

## Proposed Changes

This PR fixes the inconsistent hover color on the Masterbar.

| before | after |
|--------|--------|
| <img width="235" alt="Screenshot 2024-05-01 at 14 18 51" src="https://github.com/Automattic/wp-calypso/assets/5287479/78d8c758-bace-4253-bb72-aab73a602ea7"> | <img width="222" alt="Screenshot 2024-05-01 at 14 19 03" src="https://github.com/Automattic/wp-calypso/assets/5287479/7bc07fbe-85f6-4ea4-bb1a-cfc910723ef8"> |
| <img width="222" alt="Screenshot 2024-05-01 at 14 19 40" src="https://github.com/Automattic/wp-calypso/assets/5287479/caebf7f8-bbfe-49f5-9c13-8372ed5543f5"> | <img width="230" alt="Screenshot 2024-05-01 at 14 19 53" src="https://github.com/Automattic/wp-calypso/assets/5287479/27822df0-83a1-45a0-82ac-8ab41475be5b"> |
| <img width="224" alt="Screenshot 2024-05-01 at 14 20 17" src="https://github.com/Automattic/wp-calypso/assets/5287479/57b687e9-3321-4786-91ea-b5e978dbf18b"> | <img width="220" alt="Screenshot 2024-05-01 at 14 20 23" src="https://github.com/Automattic/wp-calypso/assets/5287479/3e80e35e-260d-4716-80f0-3fe018c6cdd1"> |
| <img width="226" alt="Screenshot 2024-05-01 at 14 20 43" src="https://github.com/Automattic/wp-calypso/assets/5287479/3b16a3cf-40d2-4677-9b96-f1c4146680d1"> | <img width="223" alt="Screenshot 2024-05-01 at 14 20 57" src="https://github.com/Automattic/wp-calypso/assets/5287479/83dcf25d-1201-42d1-b166-f39ced4bac57"> |
| <img width="230" alt="Screenshot 2024-05-01 at 14 21 18" src="https://github.com/Automattic/wp-calypso/assets/5287479/28214f13-89d2-4497-8d6e-b8b452e93c10"> | <img width="223" alt="Screenshot 2024-05-01 at 14 21 39" src="https://github.com/Automattic/wp-calypso/assets/5287479/151d90be-bb84-48d8-8e5f-8308268323cd"> |
| <img width="222" alt="Screenshot 2024-05-01 at 14 22 07" src="https://github.com/Automattic/wp-calypso/assets/5287479/e3c4fb79-8240-444b-b1f7-b03362bfd88e"> | <img width="223" alt="Screenshot 2024-05-01 at 14 22 27" src="https://github.com/Automattic/wp-calypso/assets/5287479/82161c15-40c9-4871-bfaf-1d5719c5a6f0"> |
| <img width="228" alt="Screenshot 2024-05-01 at 14 24 20" src="https://github.com/Automattic/wp-calypso/assets/5287479/67f09e1f-9786-44f3-8ce5-992cc0991ccb"> | <img width="224" alt="Screenshot 2024-05-01 at 14 24 26" src="https://github.com/Automattic/wp-calypso/assets/5287479/3f151629-af57-4717-89a8-92d54676aa63"> |
| <img width="226" alt="Screenshot 2024-05-01 at 14 24 52" src="https://github.com/Automattic/wp-calypso/assets/5287479/4c3937ea-160b-4028-8548-e0cd93a6546d"> | <img width="224" alt="Screenshot 2024-05-01 at 14 25 09" src="https://github.com/Automattic/wp-calypso/assets/5287479/c5ec5e7a-e6fa-4ff6-a55d-1aa9fd98b82e"> |
| <img width="226" alt="Screenshot 2024-05-01 at 14 25 35" src="https://github.com/Automattic/wp-calypso/assets/5287479/120bfb89-dc15-470e-afca-4a71ebd64a5d"> | <img width="221" alt="Screenshot 2024-05-01 at 14 26 14" src="https://github.com/Automattic/wp-calypso/assets/5287479/9e7e636a-ac21-4433-b7e2-24aee377df84"> |

This PR only affects color schemes from Core:

<img width="980" alt="Screenshot 2024-05-01 at 14 23 34" src="https://github.com/Automattic/wp-calypso/assets/5287479/6917155a-2f93-440b-abfb-7768fd4e42d0">

Please be aware that the color schemes in Nav Unification for Calypso were intentionally different from those in wp-admin, as discussed in this GitHub issue: https://github.com/Automattic/wp-calypso/issues/45675. Now, in the Untangling & Nav Redesign projects, we aim for consistency between Calypso and wp-admin.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Prepare an Atomin site with Classic View 
* Go to the site's `_cli` and run `wp option update wpcom_classic_early_release 1`.
* Compare the Masterbar in wp-admin and Calypso page (e.g., `/home/<your-site>`)

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

okmttdhr committed:
Fix inconsistent hover color on Masterbar

eoigal reviewed (commented):
This lgtm - this is likely a follow up but the Reader icon isn't adopting the highlight color

<img width="164" alt="Screenshot 2024-05-01 at 11 48 40" src="https://github.com/Automattic/wp-calypso/assets/5560595/5b583175-07f8-432a-b17c-f4c6c6d591ed">

Also, my atomic site in classic view (and `wpcom_classic_early_relase` enabled) isn't showing me the `All Sites` option in masterbar when I go to https://container-quizzical-einstein.calypso.live/home/anotheratomicsite3.wpcomstaging.com

allilevine reviewed (approved):
The text hover color matches when I compare wp-admin and Calypso. There are still some differences in icon hover colors, like @eoigal noted. I tested other site types:
- Atomic Classic without early release: the hover color in Calypso looks right. The hover color in wp-admin is currently incorrect:

<img width="223" alt="Screen Shot 2024-05-01 at 3 24 37 PM" src="https://github.com/Automattic/wp-calypso/assets/1689238/893d8902-198f-4fcf-8dc6-2534c928ace7">

- Atomic Default: also correct in Calypso, but there's no hover color in wp-admin.
- Simple Classic: Calypso is correct and wp-admin has the same text hover color.
- Simple Default: same as Atomic Default.
- Jetpack: Calypso and wp-admin color scheme aren't synced, but you see the correct Calypso text hover color.

fushar reviewed (approved):
Looks good!

I think we don't need to worry about the Reader icon issue. It will be present side-by-side with wp-admin icons only on Classic non-early release, which will be gone when we drop the early-release flag when we launch. (CMIIW)

okmttdhr commented:
Thank you all for testing & reviewing! 

I am about to merge this PR based on the following understanding:

> * Atomic Classic **without** early release: the hover color in Calypso looks right. The hover color in wp-admin is currently incorrect:

This is expected to be resolved with the launch of the Nav Redesign, as we will be removing the early-release flag.
And there is no need for concern regarding the Reader icon: https://github.com/Automattic/wp-calypso/pull/90136#pullrequestreview-2034745164

> * Atomic Default: also correct in Calypso, but there's no hover color in wp-admin.
> * Simple Default: same as Atomic Default.

Nav Redesign will be enabled for sites with `get_option( 'wpcom_admin_interface' ) === 'wp-admin'`, so the Reader icon will remain on the top bar for Default interface sites.
Since we changed the hover color on Calypso in this PR, 

1. We should apply the hover color to wp-admin in Default interface sites for consistency
1. OR, disable the hover color on Calypso when in Default interface sites for consistency

I'll try the latter. 

> * Jetpack: Calypso and wp-admin color scheme aren't synced, but you see the correct Calypso text hover color.

Color schemes are supposed to be synced when the site is hosted by WordPress.com (= Simple or Atomic): https://github.com/Automattic/jetpack/pull/36341

okmttdhr committed:
Consider Default interface sites

okmttdhr commented:
> > * Atomic Default: also correct in Calypso, but there's no hover color in wp-admin.
> > * Simple Default: same as Atomic Default.
> 
> Nav Redesign will be enabled for sites with `get_option( 'wpcom_admin_interface' ) === 'wp-admin'`, so the Reader icon will remain on the top bar for Default interface sites. Since we changed the hover color on Calypso in this PR,
> 
> 1. We should apply the hover color to wp-admin in Default interface sites for consistency
> 2. OR, disable the hover color on Calypso when in Default interface sites for consistency
> 
> I'll try the latter.

I updated the hover color to match Calypso and wp-admin. https://github.com/Automattic/wp-calypso/pull/90136/commits/5ae2585297bf5bc430cb01ce5fc128799d47fd36

Default interface, Calypso
<img width="132" alt="Screenshot 2024-05-02 at 13 31 06" src="https://github.com/Automattic/wp-calypso/assets/5287479/a1d36d52-950f-44a3-94cf-469ff727aec0">
<img width="134" alt="Screenshot 2024-05-02 at 13 31 10" src="https://github.com/Automattic/wp-calypso/assets/5287479/ee2f459c-2234-4ec7-83f4-c763617d970a">

Default interface, wp-admin (by switching the `View` toggle)
<img width="129" alt="Screenshot 2024-05-02 at 13 32 16" src="https://github.com/Automattic/wp-calypso/assets/5287479/35be1c7c-007d-4174-b96c-ae2ed90318c4">
<img width="130" alt="Screenshot 2024-05-02 at 13 32 20" src="https://github.com/Automattic/wp-calypso/assets/5287479/7137a66f-1a1b-442b-be82-bf75828d6079">

There is a padding difference that we should address in another PR. (This is the case in production as well)
